pyHS100/pyHS100.py

```diff
@@ -333,7 +341,18 @@ def mac(self, mac):
         :param str mac: mac in hexadecimal with colons, e.g. 01:23:45:67:89:ab
         :raises SmartPlugException: on error
         """
-        self._query_helper("system", "set_mac_addr", {"mac": mac})
+        info = self.sys_info
+
+        if 'mac' in info:
+            self._query_helper("system", "set_mac_addr", {"mac": mac})
+        elif 'mic_mac' in info:
+            self._query_helper(
+                "system",
+                "set_mac_addr",
+                {"mic_mac": mac.replace(':', '')}
```

Comment:
I think your solution is good enough, could you rebase and let's get it merged then :)